libcxx/include/atomic

```diff
@@ -705,6 +719,72 @@ _LIBCUDACXX_INLINE_VISIBILITY inline _LIBCUDACXX_CONSTEXPR int __to_gcc_failure_
               __ATOMIC_CONSUME))));
 }
 
+template<class _Tp>
+_LIBCUDACXX_INLINE_VISIBILITY inline void __raw_atomic_store(_Tp* __a, _Tp __val, int __order) {
```

Comment:
I'm trying to understand the layering here. From reading the files above (and I may revisit this when I read the changes below), it seemed to me that the `__raw_atomic_*` layer was to implement the C++ atomics backed that we plug into libc++; but now I'm also seeing it here, and the code is actually really similar to what I've seen above, only without scopes. This seems like repetition; can you explain why we need to duplicate some of these function shapes? (`__raw_atomic_store` seems fine, but the logic for `__raw_atomic_load` for instance is duplicated in our `__cxx_atomic_load` and that is somewhat concerning to me.)